File: javascript/word-count/word-count.js
export const countWords = words => {
  words = words.toLowerCase().match(/\b[\w']+\b/g)

  return words.reduce((word_count, word) => {
    word_count[word] ? word_count[word]++ : word_count[word] = 1
    return word_count
  }, {});
};

File: ruby/word-count/word_count.rb
class Phrase
  def initialize(sentence)
    @sentence = sentence
  end

  def word_count
    words.each_with_object(Hash.new(0)) { |word, hash| hash[word] += 1}
  end

  private
  attr_accessor :sentence
  MATCH_WORDS = /\b[\w']+\b/

  def words
    sentence.downcase.scan(MATCH_WORDS)
  end
end
